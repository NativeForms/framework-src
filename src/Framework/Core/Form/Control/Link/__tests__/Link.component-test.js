import React from 'react';
import { shallow } from 'enzyme';
import LinkComponent from '../Link.component';
import i18n from '../Link.i18n';

const messages = i18n.en;
const mockRoute = 'http://www.google.com';

describe('components <LinkComponent />', () => {
  it('renders correctly -- string type', () => {
    const props = {
      label: messages.title,
      route: mockRoute
    };
    const wrapper = shallow(
      <LinkComponent {...props} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('renders correctly -- object type', () => {
    const props = {
      label: {
        code: messages.googleLabel
      },
      route: mockRoute
    };
    const wrapper = shallow(
      <LinkComponent {...props} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
