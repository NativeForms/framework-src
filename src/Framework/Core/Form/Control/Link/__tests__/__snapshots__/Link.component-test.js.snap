// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <LinkComponent /> renders correctly -- object type 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  onPress={[Function]}
  underlayColor="black"
>
  <View>
    <FormattedMessage
      id="Click to visit google!"
      values={Object {}}
    />
  </View>
</TouchableHighlight>
`;

exports[`components <LinkComponent /> renders correctly -- string type 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  onPress={[Function]}
  underlayColor="black"
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      label
    </Text>
  </View>
</TouchableHighlight>
`;
